stories/control/zoom.stories.tsx

```diff
@@ -0,0 +1,89 @@
+import { Meta } from '@storybook/react';
+import { useState } from 'react';
+
+import { Axis, LineSeries, Plot, Annotations, SeriesPoint } from '../../src';
+import { Rectangle } from '../../src/components/Annotations/Rectangle';
+import { DEFAULT_PLOT_CONFIG } from '../utils';
+
+export default {
+  title: 'API/Zoom',
+} as Meta;
+
+const data = [
+  { x: 1, y: 10 },
+  { x: 2, y: 5 },
+  { x: 3, y: 3 },
+  { x: 4, y: 5 },
+  { x: 5, y: 10 },
+  { x: 8, y: 5 },
+  { x: 9, y: 1 },
+];
+
+interface Positions {
+  x1: number;
+  x2: number;
+}
+
+interface ZoomProps {
+  data: SeriesPoint[];
+  displayMarker?: boolean;
+}
+function Zoom({ data, displayMarker }: ZoomProps) {
+  const [position, setPosition] = useState<Positions | null>(null);
+  const [Click, setClick] = useState<boolean>(false);
+  const [min, setMin] = useState<number | undefined>(undefined);
+  const [max, setMax] = useState<number | undefined>(undefined);
```

Comment:
Please consolidate this to one state object (with the position and the min/max).
Move the "click" information to a `useRef()`, because the value is not used to render the chart, so it doesn't need to trigger a rerender when it changes.